### PR TITLE
fix: fixed tests due to backend changes IL-196

### PR DESCRIPTION
**Pull request by yuliahar**
**Task's name:**
[IL-229](https://jira.it-sl.ru/browse/IL-229) [QA] Fix API tests for get user by ID functionality

**Developer's full name:**
Yuliya Harbacheuskaya

**Changes:**

Fixed tests due to changes in the backend logic [IL-196](https://jira.it-sl.ru/browse/IL-196)
Removed obsolete tests.

[IL-196]: https://zufar-team.atlassian.net/browse/IL-196?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: framework/endpoints/users_api.py [hunk 1]
```diff
@@ -10,17 +10,15 @@ def __init__(self):
         self.url = HOST + "/api/v1/users"
         self.headers = {"Content-Type": "application/json"}
 
-    def get_user_by_id(self, user_id: str = "", token: str = "") -> Response:
+    def get_user_by_id(self, token: str = "") -> Response:
```
Comment (Sanya124) [inline framework/endpoints/users_api.py]: if we don't get the user by id now, then the method name is incorrect

Comment (yuliahar) [reply framework/endpoints/users_api.py]: You're right about the method naming. I agree that the current method name might be misleading. I'll update it to reflect its actual functionality more accurately. Much appreciated!

Comment (yuliahar) [reply framework/endpoints/users_api.py]: Fixed.

**Review by TetianaPerinha — APPROVED** [review]
Approve

**Review by Sanya124 — APPROVED** [review]

**Pull request merged**